proxy/proxy.go

```diff
@@ -672,13 +672,15 @@ func (c *client) handleQuery(raw *frame.RawFrame, msg *partialQuery, customPaylo
 	handled, stmt, err := parser.IsQueryHandled(parser.IdentifierFromString(c.keyspace), msg.query)
 
 	if handled {
+		c.proxy.logger.Debug("Query is handled")
 		if err != nil {
 			c.proxy.logger.Error("error parsing query to see if it's handled", zap.Error(err))
 			c.send(raw.Header, &message.Invalid{ErrorMessage: err.Error()})
 		} else {
 			c.interceptSystemQuery(raw.Header, stmt)
 		}
 	} else {
+		c.proxy.logger.Debug("Query is not handled")
```

Comment:
Maybe you added these in your testing but I'm not sure if you want to leave them? If you want to leave them then I'd consider adding some context to the log like the query string

Reply:
So there's actually already logging of the incoming query [a few lines above this](https://github.com/datastax/cql-proxy/blob/main/proxy/proxy.go#L670) but it does wind up making things pretty verbose:

```
$ ./cql-proxy -c 127.0.0.1 --bind ":9142" --debug                                                                                                                 
...
2024-06-27T17:17:59.330-0500    DEBUG   proxy/proxy.go:670      handling query  {"query": "SELECT * FROM system.peers_v2", "stream": 3}      
2024-06-27T17:17:59.330-0500    DEBUG   proxy/proxy.go:675      Query is handled    
2024-06-27T17:17:59.330-0500    DEBUG   proxy/proxy.go:670      handling query  {"query": "SELECT * FROM system.local WHERE key='local'", "stream": 4}
2024-06-27T17:17:59.330-0500    DEBUG   proxy/proxy.go:675      Query is handled    
2024-06-27T17:17:59.331-0500    DEBUG   proxy/proxy.go:670      handling query  {"query": "SELECT * FROM system.peers", "stream": 5}            
2024-06-27T17:17:59.331-0500    DEBUG   proxy/proxy.go:675      Query is handled    
2024-06-27T17:17:59.334-0500    DEBUG   proxy/proxy.go:670      handling query  {"query": "SELECT * FROM system_schema.keyspaces", "stream": 6}  
2024-06-27T17:17:59.334-0500    DEBUG   proxy/proxy.go:683      Query is not handled
2024-06-27T17:17:59.336-0500    DEBUG   proxy/proxy.go:670      handling query  {"query": "SELECT * FROM system_schema.tables", "stream": 7}   
2024-06-27T17:17:59.336-0500    DEBUG   proxy/proxy.go:683      Query is not handled
2024-06-27T17:17:59.336-0500    DEBUG   proxy/proxy.go:670      handling query  {"query": "SELECT * FROM system_schema.columns", "stream": 8} 
2024-06-27T17:17:59.336-0500    DEBUG   proxy/proxy.go:683      Query is not handled
2024-06-27T17:17:59.336-0500    DEBUG   proxy/proxy.go:670      handling query  {"query": "SELECT * FROM system_schema.types", "stream": 9} 
2024-06-27T17:17:59.337-0500    DEBUG   proxy/proxy.go:683      Query is not handled
2024-06-27T17:17:59.337-0500    DEBUG   proxy/proxy.go:670      handling query  {"query": "SELECT * FROM system_schema.functions", "stream": 10}        
2024-06-27T17:17:59.337-0500    DEBUG   proxy/proxy.go:683      Query is not handled
2024-06-27T17:17:59.337-0500    DEBUG   proxy/proxy.go:670      handling query  {"query": "SELECT * FROM system_schema.aggregates", "stream": 11}    
2024-06-27T17:17:59.337-0500    DEBUG   proxy/proxy.go:683      Query is not handled
...
```

Seems like we'd be better off merging all of that logging into a single log statement.  I'll work on making that happen.

Reply:
With the [most recent change](https://github.com/datastax/cql-proxy/pull/129/commits/c51a80fab5e2f013ef12b0a144821cbf2da0abc4) this now becomes:

```
$ ./cql-proxy -c 127.0.0.1 --bind ":9142" --debug            	 
2024-06-27T17:26:15.710-0500	DEBUG   proxycore/cluster.go:192    	connecting to cluster   {"connect timeout": "10s"}
2024-06-27T17:26:15.721-0500	INFO	proxycore/cluster.go:263    	adding host to the cluster  	{"host": "127.0.0.1:9042"}
2024-06-27T17:26:15.721-0500	DEBUG   proxycore/cluster.go:164    	control connection connected	{"endpoint": "127.0.0.1:9042"}  
2024-06-27T17:26:15.721-0500	INFO	proxy/proxy.go:376  	no local DC configured using DC from the first successful contact point {"dc": "datacenter1"}
2024-06-27T17:26:15.721-0500	DEBUG   proxycore/connpool.go:132   	creating connection pool    	{"endpoint": "127.0.0.1:9042", "connect timeout": "10s"}
2024-06-27T17:26:15.722-0500	INFO	proxy/run.go:286    	proxy is listening  	{"address": "[::]:9142"}
2024-06-27T17:26:21.134-0500	DEBUG   proxy/proxy.go:672  	Query handled by proxy  {"query": "SELECT * FROM system.peers_v2", "stream": 3}
2024-06-27T17:26:21.134-0500	DEBUG   proxy/proxy.go:672  	Query handled by proxy  {"query": "SELECT * FROM system.local WHERE key='local'", "stream": 4}
2024-06-27T17:26:21.134-0500	DEBUG   proxy/proxy.go:672  	Query handled by proxy  {"query": "SELECT * FROM system.peers", "stream": 5}
2024-06-27T17:26:21.135-0500	DEBUG   proxy/proxy.go:680  	Query not handled by proxy, forwarding  {"query": "SELECT * FROM system_schema.keyspaces", "stream": 6}
2024-06-27T17:26:21.138-0500	DEBUG   proxy/proxy.go:680  	Query not handled by proxy, forwarding  {"query": "SELECT * FROM system_schema.tables", "stream": 7}
2024-06-27T17:26:21.138-0500	DEBUG   proxy/proxy.go:680  	Query not handled by proxy, forwarding  {"query": "SELECT * FROM system_schema.columns", "stream": 8}
2024-06-27T17:26:21.138-0500	DEBUG   proxy/proxy.go:680  	Query not handled by proxy, forwarding  {"query": "SELECT * FROM system_schema.types", "stream": 9}
2024-06-27T17:26:21.138-0500	DEBUG   proxy/proxy.go:680  	Query not handled by proxy, forwarding  {"query": "SELECT * FROM system_schema.functions", "stream": 10}
2024-06-27T17:26:21.138-0500	DEBUG   proxy/proxy.go:680  	Query not handled by proxy, forwarding  {"query": "SELECT * FROM system_schema.aggregates", "stream": 11}
2024-06-27T17:26:21.138-0500	DEBUG   proxy/proxy.go:680  	Query not handled by proxy, forwarding  {"query": "SELECT * FROM system_schema.triggers", "stream": 12}
2024-06-27T17:26:21.138-0500	DEBUG   proxy/proxy.go:680  	Query not handled by proxy, forwarding  {"query": "SELECT * FROM system_schema.indexes", "stream": 13}
2024-06-27T17:26:21.138-0500	DEBUG   proxy/proxy.go:680  	Query not handled by proxy, forwarding  {"query": "SELECT * FROM system_schema.views", "stream": 14}
2024-06-27T17:26:21.138-0500	DEBUG   proxy/proxy.go:680  	Query not handled by proxy, forwarding  {"query": "SELECT * from system_virtual_schema.keyspaces", "stream": 15}
2024-06-27T17:26:21.138-0500	DEBUG   proxy/proxy.go:680  	Query not handled by proxy, forwarding  {"query": "SELECT * from system_virtual_schema.tables", "stream": 16}
2024-06-27T17:26:21.138-0500	DEBUG   proxy/proxy.go:680  	Query not handled by proxy, forwarding  {"query": "SELECT * from system_virtual_schema.columns", "stream": 17}
2024-06-27T17:26:21.158-0500	DEBUG   proxy/proxy.go:672  	Query handled by proxy  {"query": "use \"FOO\"", "stream": 2}
2024-06-27T17:26:21.159-0500	DEBUG   proxycore/connpool.go:132   	creating connection pool    	{"endpoint": "127.0.0.1:9042", "connect timeout": "10s"}
2024-06-27T17:26:21.162-0500	DEBUG   proxy/proxy.go:672  	Query handled by proxy  {"query": "USE \"FOO\"", "stream": 3}
2024-06-27T17:26:21.163-0500	DEBUG   proxy/proxy.go:680  	Query not handled by proxy, forwarding  {"query": "select * from bar", "stream": 4}
```